src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/StackOverflowClient.java

```diff
@@ -0,0 +1,136 @@
+      "https://api.stackexchange.com/2.2/answers/%s?order"
+          + "=desc&sort=activity&site=stackoverflow&filter=!9_bDE(fI5";
+  private static final int ID_INDEX = 2;
+  private static final String ITEM_PARAMETER = "items";
+  private static final String TITLE_PARAMETER = "title";
+  private static final String BODY_PARAMETER = "body";
+  private static final String CODE_PARAMETER = "code";
+  private static final String ANSWER_ID_PARAMETER = "answer_id";
+  private static final int DESCRIPTION_LENGTH_PARAMETER = 200;
+
+  public Card search(String stackLink) {
+    try {
+      Card card = getQuestion(stackLink);
+      card = getAnswerId(card);
+      card = getAnswer(card);
+      card = getAnswer(card);
+      return card;
+    } catch (URISyntaxException e) {
+      // return null card if no valid card available
+      return null;
+    }
+  }
+
+  /* Get the question id based on url from the CSE result */
+  private Card getQuestion(String stackLink) throws URISyntaxException {
+    Card card = new Card();
+    URI uri = new URI(stackLink);
+    // Parse the URL to get the question id.
+    String[] segments = uri.getPath().split("/");
+    String idStr = segments[ID_INDEX];
+    String url = String.format(SEARCH_URL_TEMPLATE, idStr);
+    try {
+      JSONObject res = getResponse(url);
+      String title =
+          res.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(TITLE_PARAMETER).toString();
+      card.setLink(uri.toString());
+      // Store the id of the question in order to get the code body of the answer.
+      card.setCode(idStr);
+      card.setTitle(title);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return card;
+  }
+
+  /* Get the most voted answer's id and store it in the card */
+  private Card getAnswerId(Card card) {
+    String url = String.format(QUESTION_URL_TEMPLATE, card.getCode());
+    try {
+      JSONObject res = getResponse(url);
+      String answerId =
+          res.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(ANSWER_ID_PARAMETER).toString();
+      // Replace the question id by the answer id in order to retrieve the code body next.
+      card.setCode(answerId);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return card;
+  }
+
+  /* Get the content of the answer and store it in the card */
+  private Card getAnswer(Card card) {
+    String url = String.format(ANSWER_URL_TEMPLATE, card.getCode());
+    try {
+      JSONObject res = getResponse(url);
+      String body =
+          res.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(BODY_PARAMETER).toString();
+      Document doc = Jsoup.parse(body);
+      // Combine all description in the answer body.
+      Elements descriptionHtml = doc.select("p");
+      String description = "";
+      for (Element e : descriptionHtml) {
+        description += e.outerHtml();
+      }
+      description = description.substring(0, DESCRIPTION_LENGTH_PARAMETER);
```

Comment:
This raises n java.lang.StringIndexOutOfBoundsException error when the description parameter is less than 200. Should add a condition where if the description is less than 200, it just sets the whole thing to the desc

Reply:
Just tested it now, still think there is an issue here with out of bounds when the description is less than 200? This is the example query/link I tested with [https://stackoverflow.com/questions/8938235/sort-an-array-in-java](https://stackoverflow.com/questions/8938235/sort-an-array-in-java)

Reply:
done, the commit was overwritten by my upstream operation